src/sinks/openobserve.rs

```diff
@@ -0,0 +1,133 @@
+use http::Uri;
+use vector_lib::codecs::encoding::{FramingConfig, JsonSerializerConfig, SerializerConfig};
+use vector_lib::configurable::configurable_component;
+
+use crate::{
+    codecs::{EncodingConfig, EncodingConfigWithFraming},
+    config::{AcknowledgementsConfig, DataType, GenerateConfig, Input, SinkConfig, SinkContext},
+    http::{Auth, MaybeAuth},
+    sinks::{
+        http::config::{HttpMethod, HttpSinkConfig},
+        util::{
+            http::RequestConfig, BatchConfig, Compression, RealtimeSizeBasedDefaultBatchSettings,
+            UriSerde,
+        },
+        Healthcheck, VectorSink,
+    },
+    tls::TlsConfig,
+};
+
+/// Configuration for the `openobserve` sink.
+#[configurable_component(sink("openobserve", "Deliver log events to OpenObserve."))]
+#[derive(Clone, Debug)]
+pub struct OpenObserveConfig {
+    /// The OpenObserve endpoint to send data to.
+    #[serde(default = "default_endpoint")]
+    #[configurable(metadata(docs::examples = "http://localhost:5080/api/default/default/_json"))]
+    uri: UriSerde,
+
+    /// The user and password to authenticate with OpenObserve endpoint.
```

Comment:
This is not accurate because `Auth` also supports token authentication.
```suggestion
    /// Authentication configuration for HTTP requests.
```